src/pages/RFOX/hooks/useLifetimeRewardsQuery.ts

```diff
@@ -31,13 +31,15 @@ export const useLifetimeRewardsQuery = ({
     () =>
       stakingAssetAccountAddress
         ? async () => {
+            // using queryClient.fetchQuery here is ok because historical epoch metadata does not change so reactivity is not needed
             const epochHistory = await queryClient.fetchQuery({
               queryKey: getEpochHistoryQueryKey(),
               queryFn: epochHistoryQueryFn,
             })
 
             const earnedByEpoch = await Promise.all(
               epochHistory.map(async epochMetadata => {
+                // using queryClient.fetchQuery here is ok because historical earnings do not change so reactivity is not needed
```

Comment:
The staleTime for `epochHistory` is actually 1 hour, so if the use happens to be needing the app to update on the exact hour an epoch. changes (ie UTC midnight on the last the of the month), they wont see this update.

Question is, should we bother to update everything to be react here and elsewhere or just deal with it?

Reply:
I guess the point is that this comment isnt entirely accurate - the epoch history changes at the beginning of every month and remains static throughout.

Reply:
((((help needed))))

Reply:
Making everything hooks *will* make things some better, though *theoretically* at the expense of overfetching in most scenarios, only to solve it for the one. So I'd be inclined to say "deal with it", but there's a but:

In *practice*, `epochHistory` is *always* fetched fresh, as there is no active query instance (escape-hatched `fetchQuery()` do contribute to the cache, but they aren't subscribers).
The only guarantee against overfetching here is in its consumers e.g `useCurrentEpochRewardsQuery` and `useLifetimeRewardsQuery`, but `epochHistory` itself will always refetch no matter what currently.

Given this fact, I don't see any reason not to move `epochHistory` to a proper `useQuery()`. We could even try to be smart and do something like this if we want to handle the end of epoch edge-case:

<img width="670" alt="image" src="https://github.com/shapeshift/web/assets/17035424/65328371-aeae-4b76-9903-9f9c3d0b9afa">